awesome_owl/static/src/todo_list/todo_list.js

```diff
@@ -0,0 +1,38 @@
+import { Component, useState } from "@odoo/owl";
+import { TodoItem } from "./todo_item";
+import { useAutofocus } from "../utils";
+
+export class TodoList extends Component {
+    static template = "awesome_owl.TodoList";
+    static components = { TodoItem };
+
+    setup() {
+        this.nextId = 1;
+        this.todos = useState([]);
+        useAutofocus("input")
+    }
+
+    addTodo(ev) {
+        if (ev.keyCode === 13 && ev.target.value != "") {
+            this.todos.push({
+                id: this.nextId++,
+                description: ev.target.value,
+                isCompleted: false
+            });
+            ev.target.value = "";
+        }
+    }
+
+    toggleTodo(todoId) {
+        const todo = this.todos.find((todo) => todo.id === todoId);
+        if (todo) {
+            todo.isCompleted = !todo.isCompleted;
+        }
+    }
+    removeTodo(todoId) {
+        const todoIndex = this.todos.findIndex((element) => element.id === todoId);
+        if (todoIndex >= 0) {
+            this.todos.splice(todoIndex, 1);
+        }
```

Comment:
`!== -1` is more explicit and widely preferred over >= 0 for findIndex checks.
```suggestion
        if (index !== -1) {
            this.todos.splice(index, 1);
        }
```